Objects/odictobject.c

```diff
@@ -1604,28 +1636,62 @@ _PyODict_SetItem_KnownHash(PyObject *od, PyObject *key, PyObject *value,
     return res;
 }
 
-int
-PyODict_SetItem(PyObject *od, PyObject *key, PyObject *value)
+static int
+_PyODict_SetItem_KnownHash(PyObject *od, PyObject *key, PyObject *value,
+                           Py_hash_t hash)
 {
+    int res;
+    Py_BEGIN_CRITICAL_SECTION(od);
+    res = _PyODict_SetItem_KnownHash_LockHeld(od, key, value, hash);
+    Py_END_CRITICAL_SECTION();
+    return res;
+}
+
+static int
+PyODict_SetItem_LockHeld(PyObject *od, PyObject *key, PyObject *value)
+{
+    _Py_CRITICAL_SECTION_ASSERT_OBJECT_LOCKED(od);
+    int res;
```

Comment:
This res is no more needed and you can directly return `_PyODict_SetItem_KnownHash_LockHeld`